server/bundles/io.cloudbeaver.server/src/io/cloudbeaver/service/session/WebSessionManager.java

```diff
@@ -144,19 +158,66 @@ public WebSession getWebSession(@NotNull HttpServletRequest request, @NotNull Ht
         return webSession;
     }
 
-    private void tryToRestorePreviousUserSession(WebSession webSession) {
-        try {
-            SMAuthInfo oldAuthInfo = webSession.getSecurityController().restoreUserSession(webSession.getSessionId());
-            if (oldAuthInfo == null) {
-                return;
+    /**
+     * Returns not expired session from cache, or restore it.
+     *
+     * @return WebSession object or null, if session expired or invalid
+     */
+    @Nullable
+    public WebSession getOrRestoreSession(@NotNull HttpServletRequest request) {
+        var httpSession = request.getSession();
+        if (httpSession == null) {
+            log.debug("Http session is null. No Web Session returned");
+            return null;
+        }
+        var sessionId = httpSession.getId();
+        WebSession webSession;
+        synchronized (sessionMap) {
+            if (sessionMap.containsKey(sessionId)) {
+                var cachedWebSession = sessionMap.get(sessionId);
+                if (!(cachedWebSession instanceof WebSession)) {
+                    log.warn("Unexpected session type: " + cachedWebSession.getClass().getName());
+                    return null;
+                }
+                return (WebSession) cachedWebSession;
+            } else {
+                try {
+                    webSession = createWebSessionImpl(httpSession);
```

Comment:
creating a web session is quite a fat entity, 
we can use only `sessionId` to know can we restore session or not, 
and if we can do it - only then create a web session